modules/base/rendering/renderabletrailtrajectory.cpp

```diff
@@ -173,30 +183,55 @@ void RenderableTrailTrajectory::deinitializeGL() {
 
 void RenderableTrailTrajectory::update(const UpdateData& data) {
     if (_needsFullSweep) {
-        // Convert the start and end time from string representations to J2000 seconds
-        _start = SpiceManager::ref().ephemerisTimeFromDate(_startTime);
-        _end = SpiceManager::ref().ephemerisTimeFromDate(_endTime);
 
-        const double totalSampleInterval = _sampleInterval / _timeStampSubsamplingFactor;
-        // How many values do we need to compute given the distance between the start and
-        // end date and the desired sample interval
-        const int nValues = static_cast<int>((_end - _start) / totalSampleInterval);
+        if (_sweepIteration == 0) {
+            // Convert the start and end time from string representations to J2000 seconds
+            _start = SpiceManager::ref().ephemerisTimeFromDate(_startTime);
+            _end = SpiceManager::ref().ephemerisTimeFromDate(_endTime);
 
-        // Make space for the vertices
-        _vertexArray.clear();
-        _vertexArray.resize(nValues);
+            _totalSampleInterval = _sampleInterval / _timeStampSubsamplingFactor;
+            _nValues = static_cast<int>((_end - _start) / _totalSampleInterval);
 
-        // ... fill all of the values
-        for (int i = 0; i < nValues; ++i) {
+            // Make space for the vertices
+            _vertexArray.clear();
+            _vertexArray.resize(_nValues);
+        }
+        
+        // Calculate sweeping range for this iteration
+        int startIndex = _sweepIteration * _sweepChunk;
+        int nextIndex = (_sweepIteration + 1) * _sweepChunk;
+        int stopIndex = (nextIndex < _nValues) ? nextIndex : _nValues;
```

Comment:
This is just: `int stopIndex = std::min(nextIndex, _nValues);`

Reply:
Indeed it is 👍 
_note to self: use std when possible_